### PR TITLE
5./api/articles

### PR DESCRIPTION
**Pull request by SmagicJones**

### CODE REVIEW

File: __tests__/news.test.js [hunk 1]
```diff
@@ -62,7 +62,17 @@ describe('GET: status 200 - responds with an article', () => {
...
+        return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then((result) => {
+                expect(Number(result.body[0].comment_count)).toBe(2)
+                expect(typeof result.body[0].topic).toBe('string')
+                expect(typeof result.body[0].title).toBe('string')
+            })
+    })
 })
```
Comment (elijaharley) [inline __tests__/news.test.js:79]: I'd consider adding a forEach here to ensure that all of the articles follow the same pattern and have the expected properties, and then when testing for a single article check the actual values of each property.

File: controller/news.controller.js [hunk 2]
```diff
@@ -26,4 +27,11 @@ exports.getArticle = (req, res) => {
...
     })
 
+}
+
+exports.getArticles = (req, res) => {
+    fetchArticles().then((result) => {
+        res.status(200).send(result)
+    })
+
 }
```
Comment (elijaharley) [inline controller/news.controller.js]: Make sure that data is being sent back with the specific key eg {result}

File: model/news.model.js [hunk 3]
```diff
@@ -22,6 +22,14 @@ exports.fetchArticle = (article_id) => {
...
-
+exports.fetchArticles = () => {
+    return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, article_img_url, COUNT(comments.article_id) AS comment_count
+    FROM articles
+    LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    GROUP BY articles.article_id;`).then((result) => {
+        return result.rows
+    })
 }
```
Comment (elijaharley) [inline model/news.model.js:35]: Nice work on this complex query.

File: playground.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, article_img_url, COUNT(comments.article_id) AS comment_count
+FROM articles
+LEFT JOIN comments
+ON articles.article_id = comments.article_id
+GROUP BY articles.article_id;
+
+
+-- SELECT COUNT(*)
+-- FROM articles
+-- WHERE votes = 0;
```
Comment (elijaharley) [inline playground.sql]: Your playground.sql & txt can go into the .gitignore - they don't need to be public

**Review by elijaharley — APPROVED** [review]
Please see my comments inline - some nice work on this endpoint, just a few tweaks needed and you're good to merge. 👍

**Review by johnhutcheon — APPROVED** [review]

**Pull request merged**